### PR TITLE
Fix PUBDEV-3682: NPE in Aggregation of tiny frame.

### PR DESCRIPTION
**Pull request by arnocandel**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/491)

### CODE REVIEW

File: h2o-algos/src/main/java/hex/aggregator/Aggregator.java [hunk 1]
```diff
@@ -36,14 +36,18 @@ public ToEigenVec getToEigenVec() {
...
+        res = new Exemplar[]{e};
+      } else {
+        int idx = es.length - 1;
+        while (es[idx] == null) idx--;
+        if (idx == es.length - 1) {
+          res = Arrays.copyOf(es, es.length << 1);
+          res[es.length] = e;
+          return res;
+        }
+        res[idx + 1] = e;
```
Comment (vpatryshev) [inline h2o-algos/src/main/java/hex/aggregator/Aggregator.java]: Will have IOOBE if es.length == 1

Comment (arnocandel) [inline h2o-algos/src/main/java/hex/aggregator/Aggregator.java]: should be fixed now, no?

**Comment by micahstubbs** [comment]
I just tested this PR with my aggregator-zoom client code.  Looks good to me!

**Comment by arnocandel** [comment]
@h2o-ops please test

**Pull request merged**